modules/imgimport/src/csvreading.cpp

```diff
@@ -0,0 +1,169 @@
+#include "csvreading.h"
+#include <fstream>
+#include <string>
+#include "metadatalinkedlist.h"
+#include <vector>
+#include <iostream>
+#include "metadata.h";
+using namespace std;
+
+Metadatalinkedlist* readcsv(const char* filename){
+    Metadatalinkedlist* result=new Metadatalinkedlist;
+    Metadatalinkedlist* poi=new Metadatalinkedlist;
+    poi=result;
```

Comment:
This is leaking memory. You're immediately discarding the Metadatalinkedlist instantiated at the previous line.
Also, why create nodes here anyway? The nodes that contain the data are created inside the while loop and the result node's only purpose is to point to the "real" head of the list. Then at the end this result node is discarded and only the rest of the list is returned. You should be able to rewrite this with result and poi both starting as null.